### ISSUE TITLE
New feature: allow admins to remove e-mails from archives

### ISSUE THREAD

**Issue by ilgrosso** [apache/incubator-ponymail#59]
Would it be of interest the possibility to give administrators the possibility to hide / remove specific e-mails from archives?

**Comment by Humbedooh** [comment]
step 1: how would you define an admin? :)
It could be done, but would require some rewrite of the elastic library to support DELETE operations

**Comment by PierreSmits** [comment]
How will we ensure that this is inline with ASF policies and procedures regarding mailing lists and postings?

**Comment by Humbedooh** [comment]
Pierre, this has no relevance to the subject. How the ASF decides to utilize an implementation of this project is their business. It has no impact on how we decide to implement new features.

**Comment by ilgrosso** [comment]
@Humbedooh maybe this is not necessarily a feature to expose via web UI, it could very well be something accessible via CLI (as other tools including importer): in this way you can safely jump the "step 1" above :-)

**Comment by Humbedooh** [comment]
edit-list.py could be enhanced to do this, I suppose.

**Comment by Humbedooh** [comment]
Emails can be hidden by deleting the digest. This is what edit-lists.py does currently. The original email is retained in the source index, but the digest is removed, thus making the email not show up on lists.

**Comment by ilgrosso** [comment]
+1 to administratively hiding e-mails: are you saying that this issue can be just closed? Is the feature already documented?

**Comment by Humbedooh** [comment]
no, the issue can't be closed just yet. You can't delete emails by ID, only mass-remove lists with that script.

**Comment by ShaneCurcuru** [comment]
We need to clarify documentation, to be specific as to what happens to the data and the user access thereto after using this (new) feature.

That is: If an "admin" uses this feather, er, feature to delete the digest of a single message, then:

- Users can no longer see it in list views.
- Users will see - what? - if they hit a URL they've previously saved that pointed at the message?
- If the message is in the middle of a thread, how does it affect threaded display?
- Does this actually delete the >quoted content from other emails in the thread? (I'd presume not, meaning we need to clearly document that, lest someone think this is a secure feature to scrub data)
- What happens if someone else wants to mirror the ponymail dataset?  I.e. after you use this feature, and I run a script to suck the config from the server to my local machine, will I get all data, or will this message be missing from my end?
- Does this survive server restarts/reconfigs?

Apologies if some of those are naive questions, but if we're gonna add something that makes people *think* they deleted a message, we need to be clear exactly how it works.
